Khan/wonder-blocks
# WB-1068 - Add Validation and Event Props to LabeledTextField

## Summary:
Added all props dealing with Validation and Event Handling to `LabeledTextField`.

The following were the props added:
* validation
* onValidation
* onChange
* onKeyDown
* onFocus
* onBlur

Added unit tests to ensure the new props work as expected.

Issue: https://khanacademy.atlassian.net/browse/WB-1068

## Test plan:
1. Open Styleguidist or React Storybook (see generated links).
2. View the `LabeledTextField` examples under "Form / LabeledTextField".
3. View at all the new examples that show validation and events in action.
4. View the examples that provide error message on validation errors.
5. Test all examples by pressing "Enter" and notice how the TextField blurs.

---

#### Normal
<img width="379" alt="labeled-text-field-normal" src="https://user-images.githubusercontent.com/60367213/122300374-db1db080-cec4-11eb-9af8-ff147a602004.png">

#### Validated
<img width="378" alt="labeled-text-field-validated" src="https://user-images.githubusercontent.com/60367213/122300393-e07afb00-cec4-11eb-8fd7-559fcce7001f.png">
